### ISSUE TITLE
Let's make a better installation script

### ISSUE THREAD

**Issue by evrifaessa** [boxbilling/boxbilling#709]
Here, I'll be dumping my ideas for the installation process. Feel free to comment yours too 👍🏻 
* Integrate some of the system settings directly to the installer
** Allow people to define the name of the website right in the installation, instead of initializing websites with the "Company Name" placeholder.
** Allow people to select timezone
** Allow people to define the support e-mail
** Allow people to define the company support number
-----
more suggestions to come soon.

**Comment by fordnox** [comment]
PR are always welcome

**Comment by rndevuk** [comment]
A suggestion for the "new Installer" instead of running sql from a string/file... maybe we should create migration files.

That way upgrades and changes to the installation will be easier to manage in the future.

**Comment by rndevuk** [comment]
Why has this been closed? has it been implemented and if so how come its not been referenced?

**Comment by evrifaessa** [comment]
> Why has this been closed? has it been implemented and if so how come its not been referenced?

No, didn't implement anything yet. I thought this would fit better into the "discussions" tab, but nvm. Reopening as requested.

**Comment by timothygwebb** [comment]
@rndevuk @evrifaessa @fordnox 

Thoughts for discussion:

I agree with @rndevuk regarding migration scripts. Please read below article for further discussion.
https://www.red-gate.com/simple-talk/sql/database-administration/using-migration-scripts-in-database-deployments/

I agree with @evrifaessa until work is begun on this topic of discussion an issue should not be created but should be converted to a discussion until a solution is documented and worked in a project.

I propose to convert this to a discussion. While discussion is active address database deployments using CI.
Please refer to the following article that is directly related to the type of database currently in use MYSQL.
https://dev.mysql.com/doc/mysql-secure-deployment-guide/5.7/en/secure-deployment-overview.html
https://dev.mysql.com/doc/mysql-secure-deployment-guide/5.7/en/secure-deployment-configuration-file.html
https://docs.gitlab.com/ee/topics/autodevops/

@fordnox  @evrifaessa  @rndevuk  if you all are in agreement I will convert this to a discussion and continue conversation in a discussion thread and create a project and tickets as work is completed that will allow the project to grow in stages and be a very thorough thought through managed process. This is not just changing the installation script this is developing a strategy for continuous integration and development for ease of future install and upgrades. Please let me know your thoughts and state to me if you agree in converting this to a discussion.
